arangod/Aql/TraversalExecutor.cpp

```diff
@@ -226,7 +232,6 @@ auto TraversalExecutor::produceRows(AqlItemBlockInputRange& input, OutputAqlItem
 
   while (true) {
     if (_traverser.hasMore()) {
-      TRI_ASSERT(_inputRow.isInitialized());
```

Comment:
Shouldn't that assertion hold again, now that the traverser is reset properly? I'm thinking we need to have an input row before we can start traversing on anything.

Reply:
well I deleted this one, because `doOutput` has the same assertion as well; maybe I just put it back in there, too. It's not as if it costs anything

Reply:
I sometimes like such "unnecessary" assertions to document invariants to the reader.